src/modules/Profile.rb

```diff
@@ -741,7 +741,8 @@ def ReadXML(file)
         @current = XML.XMLToYCPFile(file)
       end
 
-      if @current != {} && Builtins.size(@current) == 0
+      xml_error = XML.XMLError
+      if xml_error && !xml_error.empty?
```

Comment:
hmm, as this is quite sensive area is possible to have a test case for it with example invalid profile?

Reply:
It is basically a backport and I've manually tested it. But you are right, I will add a test case.